lib/traject/macros/stanford.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+module Macros
+  # Macros for extracting Stanford Specific MODS values from Nokogiri documents
+  module Stanford
+    NS = { mods: 'http://www.loc.gov/mods/v3',
+           rdf: 'http://www.w3.org/1999/02/22-rdf-syntax-ns#',
+           dc: 'http://purl.org/dc/elements/1.1/' }.freeze
+    # SOURCE SPECIFIC METHODS
+    # Macros for extracting Stanford-specific values from Nokogiri documents
+    def generate_sul_manifest
+      lambda { |record, accumulator, context|
+        druid = generate_druid(record, context)
+        # eventually add a test that calls IIIF service to check
+        if druid.present? && druid?(druid.chomp('druid:'))
+          manifest = "https://purl.stanford.edu/#{druid}/iiif/manifest"
+          accumulator << manifest
+          grab_sul_iiif_links(manifest)
+          generate_sul_shown_at(druid)
+        end
+      }
+    end
+
+    def generate_druid(record, context)
```

Comment:
I think `generate_druid` and `druid?` (and probably some other internal methods) can/shuld be marked as private?